core/domain/exp_jobs_one_off.py

```diff
@@ -216,68 +256,133 @@ def map(item):
             for state_name, state in exploration.states.items():
                 if state.interaction.id == 'MathExpressionInput':
+                    types_of_inputs = set()
                     for group in state.interaction.answer_groups:
                         for rule_spec in group.rule_specs:
                             rule_input = ltt.latex_to_text(
                                 rule_spec.inputs['x'])
 
-                            # Shifting powers in trig functions to the end.
-                            # For eg. 'sin^2(x)' -> 'sin(x)^2'.
-                            for trig_fn in trig_fns:
-                                rule_input = re.sub(
-                                    r'%s(\^\d)\((.)\)' % trig_fn,
-                                    r'%s(\2)\1' % trig_fn, rule_input)
-
-                            # Adding parens to trig functions that don't have
-                            # any. For eg. 'cosA' -> 'cos(A)'.
-                            for trig_fn in trig_fns:
-                                rule_input = re.sub(
-                                    r'%s(?!\()(.)' % trig_fn,
-                                    r'%s(\1)' % trig_fn, rule_input)
-
-                            # The pylatexenc lib outputs the unicode values of
-                            # special characters like sqrt and pi, which is why
-                            # they need to be replaced with their corresponding
-                            # text values before performing validation.
-                            for unicode_char, text in (
-                                    unicode_to_text_mapping.items()):
-                                rule_input = rule_input.replace(
-                                    unicode_char, text)
-
-                            # Replacing trig functions that have format which is
-                            # incompatible with the validations.
-                            for invalid_trig_fn, valid_trig_fn in (
-                                    inverse_trig_fns_mapping.items()):
-                                rule_input = rule_input.replace(
-                                    invalid_trig_fn, valid_trig_fn)
+                            rule_input = clean_math_expression(rule_input)
 
                             validity = 'Invalid'
                             if is_valid_math_expression(rule_input):
-                                validity = 'Valid Expression'
+                                validity = 'Valid Algebraic Expression'
+                            elif is_valid_math_expression(
+                                    rule_input, algebraic=False):
+                                validity = 'Valid Numeric Expression'
                             elif is_valid_math_equation(rule_input):
-                                validity = 'Valid Equation'
+                                validity = 'Valid Math Equation'
+
+                            if validity != 'Invalid':
```

Comment:
If you use a string more than once, always make it a named constant.

Reply:
Done.